lib/task-service.js

```diff
@@ -4,21 +4,54 @@
 
 var di = require('di');
 
-module.exports = factory;
-di.annotate(factory, new di.Provide('Services.Task'));
-di.annotate(factory, new di.Inject(
+module.exports = taskServiceFactory;
+di.annotate(taskServiceFactory, new di.Provide('Services.Task'));
+di.annotate(taskServiceFactory, new di.Inject(
+    'Logger',
+    'Promise',
+    'ChildProcess',
     'TaskOption.Validator'
 ));
 
-function factory(validator) {
+function taskServiceFactory(
+    Logger,
+    Promise,
+    ChildProcess,
+    validator
+) {
+    var logger = Logger.initialize(taskServiceFactory);
+    var requiredCmdTools = ['ipmitool'];
```

Comment:
@brianparry : good idea. So seems it's better to annotate some required tools for each job or task or graph, when that graph or task is triggered, we can first scan the existence of those tools and fail immediately if any tool is missing, does this match your thinking?